Sources/ArgumentParser/Parsable Properties/Flag.swift

```diff
@@ -196,16 +223,84 @@ extension Flag where Value == Bool {
     inversion: FlagInversion,
     exclusivity: FlagExclusivity = .chooseLast,
     help: ArgumentHelp? = nil
   ) {
-    self.init(_parsedValue: .init { key in
-      .flag(key: key, name: name, default: initial, inversion: inversion, exclusivity: exclusivity, help: help)
-      })
+    self.init(
+      name: name,
+      initial: initial,
+      inversion: inversion,
+      exclusivity: exclusivity,
+      help: help
+    )
+  }
+
+  /// Creates a Boolean property with default value provided by standard Swift default value syntax that reads its value from the presence of one or more inverted flags.
+  ///
+  /// Use this initializer to create a Boolean flag with an on/off pair.
+  /// With the following declaration, for example, the user can specify either `--use-https` or `--no-use-https` to set the `useHTTPS` flag to `true` or `false`, respectively.
+  ///
+  /// ```swift
+  /// @Flag(inversion: .prefixedNo)
+  /// var useHTTPS: Bool = true
+  /// ````
+  ///
+  /// - Parameters:
+  ///   - name: A specification for what names are allowed for this flag.
+  ///   - wrappedValue: A default value to use for this property, provided implicitly by the compiler during propery wrapper initialization.
+  ///   - inversion: The method for converting this flag's name into an on/off pair.
+  ///   - exclusivity: The behavior to use when an on/off pair of flags is specified.
+  ///   - help: Information about how to use this flag.
+  public init(
+    wrappedValue: Bool,
+    name: NameSpecification = .long,
+    inversion: FlagInversion,
+    exclusivity: FlagExclusivity = .chooseLast,
+    help: ArgumentHelp? = nil
+  ) {
+    self.init(
+      name: name,
+      initial: wrappedValue,
+      inversion: inversion,
+      exclusivity: exclusivity,
+      help: help
+    )
+  }
+
+  /// Creates a Boolean property with no default value that reads its value from the presence of one or more inverted flags.
+  ///
+  /// Use this initializer to create a Boolean flag with an on/off pair.
+  /// With the following declaration, for example, the user can specify either `--use-https` or `--no-use-https` to set the `useHTTPS` flag to `true` or `false`, respectively.
+  ///
+  /// ```swift
+  /// @Flag(inversion: .prefixedNo)
+  /// var useHTTPS: Bool
+  /// ````
+  ///
+  /// - Parameters:
+  ///   - name: A specification for what names are allowed for this flag.
+  ///   - wrappedValue: A default value to use for this property, provided implicitly by the compiler during propery wrapper initialization.
+  ///   - inversion: The method for converting this flag's name into an on/off pair.
+  ///   - exclusivity: The behavior to use when an on/off pair of flags is specified.
+  ///   - help: Information about how to use this flag.
+  public init(
+    name: NameSpecification = .long,
+    inversion: FlagInversion,
+    exclusivity: FlagExclusivity = .chooseLast,
+    help: ArgumentHelp? = nil
+  ) {
+    self.init(
+      name: name,
+      initial: false,
```

Comment:
@natecook1000 I just want to point out one thing here that we should figure out how to resolve before this gets merged. I've preserved as much of the existing intent as possible, but the old `@Flag(default: nil)` is no longer possible now on this commit. I personally think it'd be better to remove the implicit default-false here:

```diff
-      initial: false,
+      initial: nil,
```

but that would necessarily mean that there would have to be source changes like this:

```diff
 @Flag(inversion: .prefixedNo)
+var someFlag: Bool
-var someFlag: Bool = false
```

or else anything that assumed that this would have some value when no flag was given would break at parse-time.

Do you have any thoughts here?

Reply:
Let me rephrase some of that for clarity:

The actual source breakage is relatively unimportant since we're already deprecating the old method, but the intent of what to do with a `Flag` if completely uninitialized (by either default value or parsing) is what I want to make sure I understand. I personally feel that an implicit default on a non-`Optional` property is less Swifty than making someone specify an explicit default as a fallback just as you would see in a function call, for instance. Forcing users to specify a default would also resolve another inconsistency I see, that `@Flag(default: nil) var someFlag: Bool` looks on a high level like it would be attempting to assign a `nil` to a non-`Optional` property; this is also compounded by this behavior not being valid for the other argument types. In effect, `nil` is being overloaded here in the public-facing API to be a selector for different parsing behavior, which feels relatively un-Swifty to me.

All of that above leads me to feel that we should change the behavior of `Flag` to have no default value in the public API without the user explicitly setting one (which would be setting the default to `nil` internally). I do realize, though, that changing this behavior will both affect existing users and create a potential minor usability hit for new users, since the most common case definitely is for a flag to be default-false, and would definitely appreciate hearing your opinion as to if that’s a good trade-off.

Reply:
Great point about the flag initializers. The intent of the existing design is to handle the two different kinds of Boolean flags, both without an inversion and with an inversion. The tricky part is these act a bit differently — here's what the semantics are, and why the current design is what it is:

_No inversion:_ This kind of flag is always optional (for the user) and defaults to `false`; if the user provides the flag, it toggles to `true`. If you were to specify a default of `true`, the flag would never have any effect, so we removed that parameter from the no-inversion `init`. Likewise, if the flag were required, it would also have no effect, since the resulting property would always be `true`.

_With inversion:_ The user can explicitly set this flag to either `true` or `false`, so the default can be either of these, or left out entirely (i.e. `nil`). If the default is `nil`, then one of the two flags is required; if there's a default value, then the flags are optional for the user.

With all that said, what should flags look like with this new way of initializing properties? I agree with you that once we're doing regular Swift default initialization, having uninitialized properties automatically get a value is weird. I think the kinds of declarations we would want to support would be:

```swift
// No inversion
@Flag var regular: Bool = false

// Inversion
@Flag(inversion: .prefixedNo) var required: Bool
@Flag(inversion: .prefixedNo) var defaultTrue: Bool = true
@Flag(inversion: .prefixedNo) var defaultFalse: Bool = false

// Unfortunately, you can also write:
@Flag var alwaysTrue: Bool = true
```

I think we could print a warning on that last case, recommending that they change the default to `false` or provide an inversion. For initializers, we'll want these two, plus any deprecated ones to maintain source-compatibility:

```swift
// cases 1, 3, 4, 5
init(wrappedValue: Bool, inversion: FlagInversion? = nil, ...)
// case 2
init(inversion: FlagInversion, ...)
```

Reply:
I think the warning for the `alwaysTrue` case is probably the best way to go; it could also be a parsing error instead, but I can see a use for allowing flags to be pinned to a value for quick testing purposes. In the absolutely pie-in-the-sky ideal scenario it seems best (to me) to have the no-inversion case be represented by something other than a boolean, something wrapped in an `Optional` that is only capable of having a single value. I'm thinking of something akin to:
```swift
enum SingleValue {
    case some
}
@Flag someFlag: SingleValue?
```
That preserves the ability to leave off a default for that case (since it'll default to `nil`), and then allow the default for `Bool` flags to be `inversion: .prefixedNo` and also just represents the high-level usage more (flag present is some value, flag not present is no value). But I realize that will probably not be incredibly intuitive for users to deal with, and opens up further questions of how to easily communicate how to migrate existing code.

So, I think the best next steps for me are pretty much exactly what you have described above:
1. Extend support for default values to the no-inversion case (which is not currently in this PR), with a warning if that value is set to `false`
    - This should be accomplished by combining those initializers, as you indicated
2. Deprecate not having a default value on no-inversion flags in favor of making the user provide an explicit `= false`, but keep it around for now to not completely break existing source
3. Change the intent of flags with inversions such that they no longer default to `false` but to being required if no default value is provided, as in your example above
    - We unfortunately won't be able to provide any sort of deprecation message or other warnings for this, but I also don't know how large a concern that is at this point.

Do you have any other thoughts? I can probably have all that implemented in the next day or so otherwise.

Reply:
> But I realize that will probably not be incredibly intuitive for users to deal with, and opens up further questions of how to easily communicate how to migrate existing code.

Exactly, this would more accurately model the correct behavior, but at the cost of making the library harder to use.

> 1. Extend support for default values to the no-inversion case (which is not currently in this PR), with a warning if that value is set to `false`
> - This should be accomplished by combining those initializers, as you indicated

I think you mean warning on `true`, but yes on this point. 👍 

> 2. Deprecate not having a default value on no-inversion flags in favor of making the user provide an explicit = false, but keep it around for now to not completely break existing source
> 3. Change the intent of flags with inversions such that they no longer default to false but to being required if no default value is provided, as in your example above
> - We unfortunately won't be able to provide any sort of deprecation message or other warnings for this, but I also don't know how large a concern that is at this point.

This is unfortunate — I hadn't remembered that leaving the default off of a flag with an inversion keeps the default as `false`. We want to change the semantics of that same line of code to be a required flag, but I don't think there's a way to provide a deprecation warning to guide users to the right syntax, which means this will necessarily be a breaking change. I think your plan forward makes sense — I'll play around with a couple different options to communicate the migration path (which may just be a statement in the changelog).

Reply:
> I think you mean warning on true, but yes on this point. 👍

Whoops my bad there, I definitely did mean `true`.

All of this is implemented in 83a70e4 and e7397e9/